tests/Adapter/Http/FileHttpClient.php

```diff
@@ -42,17 +42,19 @@ public function request($method, $path, $body = null, array $options = [])
     public function asyncRequest($method, $path, $body = null, array $options = [])
     {
         $filePath = $this->fileResolver->getFilePath($method, $path, $body, $options);
-        return new Promise(
-            function () use ($filePath) {
+        /** @var PromiseInterface $promise */
+        $promise = new Promise(
+            function () use (&$promise, $filePath) {
```

Comment:
I don't think you need it to be used as a reference

Reply:
I don't think either, but it was used like this in the [Guzzle promise documentation](https://github.com/guzzle/promises#synchronous-wait).

Reply:
It must be a limitation of the PHP language. If I remove the `&` operator, I get a notice error:
```
Notice: Undefined variable: promise in tests/Adapter/Http/FileHttpClient.php line 47
```